tracer/src/Datadog.Trace/Agent/TracesTransportStrategy.cs

```diff
@@ -15,6 +15,7 @@ namespace Datadog.Trace.Agent
     internal static class TracesTransportStrategy
     {
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<Tracer>();
+        private static readonly TimeSpan Timeout = TimeSpan.FromSeconds(15);
```

Comment:
I understand using 15s here, as that's what we used for telemetry and logs, but I wonder if we could (should) make it shorter than that.
 
Logs and telemetry are being sent to an external API, whereas the traces are (in most cases) being sent to a "local" address of some sort. Given we have 5(?) retries, is it worth making this timeout shorter?

Also, given we already expose `TracesPipeTimeoutMs`, is it worth exposing this timeout publicly (even if only publicly as an environment variable) in case customers want to bump it up/reduce it?

Reply:
Yeah reducing it makes sense. Is 5s okay?

We could expose it publicly but I don't feel a big motivation to do so right now, as we haven't gotten such a request before 🤷🏼

Reply:
> Is 5s okay

Makes sense to me, but would wonder what other people think?

> We could expose it publicly but I don't feel a big motivation to do so right now, as we haven't gotten such a request before

My only concern is that _someone_ will be impacted by us using the shorter timeout, and if we have the public knob we can tell them to increase it as necessary?

Reply:
Ah that's a good point about breaking someone. Ok let me whip something up in this same PR

Reply:
@andrewlock I added an environment variable config in 2b0e27511370f18142a6b59ab795a970b63904e3

Reply:
I kept the default at 15s since the agent isn't always on the same host and it can be tuned downward by the user if needed